### PR TITLE
Make RPG Maker 2003 battle hardcoded terms customizable

### PR DESCRIPTION
**Pull request by rueter37**
This PR makes the hardcoded terms in RPG Maker 2003 battle customizable and adds skill and item start messages in RPG Maker 2003 battles.

Fixes: #2362

### CODE REVIEW

File: src/battle_message.cpp [hunk 1]
```diff
@@ -432,36 +440,110 @@ std::string GetItemStartMessage2k(const Game_Battler& source, const lcf::rpg::It
...
+		lcf::Data::terms.easyrpg_battle2k3_double_attack,
+		Utils::MakeArray('S'),
+		Utils::MakeSvArray(source.GetName())
+	);
 }
 
-std::string GetSkillStartMessage2k3(const lcf::rpg::Skill& skill) {
-	return ToString(skill.name);
+std::string GetSkillStartMessage2k3(const Game_Battler& source, const Game_Battler* target, const lcf::rpg::Skill& skill) {
+	StringView target_name = "???";
```
Comment (Ghabry) [inline src/battle_message.cpp:455]: As these placeholders are from 2kE: When you use a skill that targets the entire party is the %O also returning ??? there?

**Review by Ghabry — APPROVED** [review]
Feels a bit strange to see placeholders which is a 2kE feature in 2k3 code but considering these are EasyRPG extensions it makes sense to use them. :)
Code-wise I see no problems here and this all implements stuff I consider useful.

**Review by fdelapena — APPROVED** [review]

**Review by fmatthew5876 — APPROVED** [review]

File: src/battle_message.cpp (hunk 1 above)
Comment (rueter37) [reply src/battle_message.cpp:455]: Tested with RPG_RT 2kE and confirmed that %O returns ??? if a skill targets the entire party.

**Review by carstene1ns — APPROVED** [review]

**Comment by carstene1ns** [comment]
While we are at it, these messages ("Double Attack", "Observe",...) should be added somehow to the translation feature without using custom terms.

**Comment by Ghabry** [comment]
At least lcftrans and translation already supports this because it is handled by ForEachString.
For the database it needs our editor 😁

**Pull request merged**